### PR TITLE
Auth and Footer

### PR DESCRIPTION
**Pull request by iamharnad**
Added Footer, Firebase authentication and css changes.

### CODE REVIEW

File: src/CheckoutProduct.css [hunk 1]
```diff
@@ -9,21 +9,34 @@
...
+button:focus {
+  background: linear-gradient(117deg, rgb(248, 137, 73), rgb(248, 82, 49));
+  box-shadow: 0 0.1rem 0.8rem rgb(254, 100, 70);
 }
 
 .checkoutProduct__image {
   object-fit: contain;
-  width: 180px;
+  width: 200px;
   height: 180px;
```
Comment (shivanigangadharan) [inline src/CheckoutProduct.css]: suggestion to use rem here instead of px

File: src/Footer.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  Row,
+  Column,
+  FooterLink,
+  Heading,
+} from './FooterStyles';
+
+const Footer = () => {
+  return (
+    <Box>
+      <h1 style={{ color: '#ff9f00', textAlign: 'center', marginTop: '-50px' }}>
```
Comment (shivanigangadharan) [inline src/Footer.js]: avoid inline styling and use class names

File: src/Footer.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        eshop: your favourite online store
+      </h1>
+      <br></br>
```
Comment (shivanigangadharan) [inline src/Footer.js]: avoid the use of `<br>` tags and give spacing through css instead.

File: src/FooterStyles.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  }
+`;
+
+export const Container = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  max-width: 1000px;
+  margin: 0 auto;
+  /* background: red; */
```
Comment (shivanigangadharan) [inline src/FooterStyles.js]: remove unwanted comments

File: src/Login.css [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  width: 98%;
+}
+
+.login__container > p {
+  margin-top: 15px;
+  font-size: 12px;
+}
+
+.login__signInButton {
+  background: #ff9f00;
```
Comment (shivanigangadharan) [inline src/Login.css]: try to use css variables whenever defining new colors to maintain consistency in ui

**Review by shivanigangadharan — COMMENTED** [review]
Suggested a few changes, can look into them. Rest of the code is good.
Also, do resolve the conflicts and fix the deploy preview it doesn't seem to work.

File: src/CheckoutProduct.js [hunk 6]
```diff
@@ -1,12 +1,22 @@
...
-const CheckoutProduct = ({ id, image, title, price, rating }) => {
+function CheckoutProduct({ id, image, title, price, rating }) {
+  const [{ basket }, dispatch] = useStateValue();
+
+  const removeFromBasket = () => {
+    dispatch({
+      type: 'REMOVE_FROM_BASKET',
+      id: id,
+    });
+  };
```
Comment (ShekharDhangar) [inline src/CheckoutProduct.js]: here you could remove the brackets after fat arrow as and all code will be in 1 line.

File: src/Footer.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            <FooterLink href='#'>Buy From Us</FooterLink>
+            <FooterLink href='#'>Vendor Management</FooterLink>
+            <FooterLink href='#'>Gurantee Period</FooterLink>
+          </Column>
+          <Column>
+            <Heading>Contact Us</Heading>
+            <FooterLink href='#'>Uttar Pradesh</FooterLink>
+            <FooterLink href='#'>Ahemdabad</FooterLink>
+            <FooterLink href='#'>Indore</FooterLink>
+            <FooterLink href='#'>Mumbai</FooterLink>
```
Comment (ShekharDhangar) [inline src/Footer.js]: You could store all this items in various arrays and then map over it.

File: src/Footer.js [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+              </i>
+            </FooterLink>
+            <FooterLink href='#'>
+              <i className='fab fa-twitter'>
+                <span style={{ marginLeft: '10px' }}>Twitter</span>
+              </i>
+            </FooterLink>
+            <FooterLink href='#'>
+              <i className='fab fa-youtube'>
+                <span style={{ marginLeft: '10px' }}>Youtube</span>
```
Comment (ShekharDhangar) [inline src/Footer.js]: Try to avoid inline styling

File: src/Login.js [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+import './Login.css';
+import { useState } from 'react';
+import StorefrontIcon from '@mui/icons-material/Storefront';
+import { Link, useHistory } from 'react-router-dom';
+import { auth } from './firebase';
+
+const Login = () => {
+  const history = useHistory();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
```
Comment (ShekharDhangar) [inline src/Login.js]: you could use 1 object instead of 2 useState.

**Review by ShekharDhangar — APPROVED** [review]
Look at the sujesstions, Rest looks good.

File: src/CheckoutProduct.css [hunk 10]
```diff
@@ -9,21 +9,34 @@
...
 }
 
 .checkoutProduct__rating {
   display: flex;
```
Comment (anushasunil) [inline src/CheckoutProduct.css]: can make utility classes for flex property

File: src/FooterStyles.js [hunk 11]
```diff
@@ -0,0 +1,58 @@
+import styled from 'styled-components';
+
+export const Box = styled.div`
+  padding: 80px 60px;
+  background: #131921;
+  position: bottom;
```
Comment (anushasunil) [inline src/FooterStyles.js]: not sure about property-value as `bottom`, assuming you intended to use either `absolute` or `fixed` property-value

File: src/Login.css [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+  border: 1px solid lightgray;
+  padding: 20px;
+}
+
+.login__container > h1 {
+  font-weight: 500;
+  margin-bottom: 20px;
+}
+
+.login__container > form > h5 {
```
Comment (anushasunil) [inline src/Login.css]: liked the use of combinators, helped in understanding

**Review by anushasunil — APPROVED** [review]
Few suggestions and also would like to highlight to create utility classes and variables in future to improve readability and maintaining consistency throughout the project. Overall, Great work.

**Pull request merged**